### PR TITLE
MULE-18336: Remove references to `exception-strategy` in app parser

### PR DESCRIPTION
**Pull request by elrodro83**

### CODE REVIEW

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/spring/ExceptionStrategyRefBeanDefinitionCreator.java [hunk 1]
```diff
@@ -1,37 +0,0 @@
...
-/**
- * Processor of the chain of responsibility that knows how to create the
- * {@link org.springframework.beans.factory.config.BeanDefinition} for an exception strategy reference element.
- *
- * @since 4.0
- */
-class ExceptionStrategyRefBeanDefinitionCreator extends BeanDefinitionCreator {
-
-  @Override
-  boolean handleRequest(CreateBeanDefinitionRequest createBeanDefinitionRequest) {
```
Comment (gsfernandes) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/spring/ExceptionStrategyRefBeanDefinitionCreator.java:27]: Where is this logic now?

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/ApplicationModel.java [hunk 2]
```diff
@@ -1041,19 +1036,6 @@ private ComponentModel getOnErrorComponentModel(ComponentModel onErrorModel) {
     }
   }
 
-  private void validateExceptionStrategyWhenAttributeIsOnlyPresentInsideChoice() {
```
Comment (gsfernandes) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/ApplicationModel.java:1044]: Who is doing this validation now?

Comment (elrodro83) [reply modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/ApplicationModel.java:1044]: this was leftovers from mule 3, mule 4 uses error-handler, not exception strategy

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/spring/ExceptionStrategyRefBeanDefinitionCreator.java (hunk 1 above)
Comment (elrodro83) [reply modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/spring/ExceptionStrategyRefBeanDefinitionCreator.java:27]: this was leftovers from mule 3, mule 4 uses error-handler, not exception strategy

**Pull request merged**